devtools/gamma-server/test/gamma_server_and_cli_test.go

```diff
@@ -0,0 +1,276 @@
+	TransactionStatus  int
+	BlockHeight        int
+	BlockTimestamp     int
+}
+
+type OutputArgumentCliResponse struct {
+	Name  string
+	Type  string
+	Value interface{}
+}
+
+type callMethodCliResponse struct {
+	OutputArguments []OutputArgumentCliResponse
+	CallResult      int
+	BlockHeight     int
+	BlockTimestamp  int
+}
+
+func ClientBinary() []string {
+	ciBinaryPath := "/opt/orbs/gamma-cli"
+	if _, err := os.Stat(ciBinaryPath); err == nil {
+		return []string{ciBinaryPath}
+	}
+
+	return []string{"go", "run", "../../gammacli/main/main.go"}
+}
+
+func runCommand(command []string, t *testing.T) string {
+	cmd := exec.Command(command[0], command[1:]...)
+	var stdout, stderr bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+	err := cmd.Run()
+
+	fmt.Println("jsonapi exec command:", command)
+	fmt.Println("command stdout:", stdout.String())
+	fmt.Println("command stderr:", stderr.String())
+
+	require.NoError(t, err, "gamma cli command should not fail")
+
+	return stdout.String()
+}
+
+func generateTransferJSON(amount uint64, targetAddress []byte) []byte {
+	// Encode the address as hex
+	hexTargetAddress := hex.EncodeToString(targetAddress)
+
+	transferJSON := &gammacli.JSONTransaction{
+		ContractName: "BenchmarkToken",
+		MethodName:   "transfer",
+		Arguments: []gammacli.JSONMethodArgument{
+			{Name: "amount", Type: "uint64", Value: amount},
+			{Name: "targetAddress", Type: "bytes", Value: hexTargetAddress},
+		},
+	}
+
+	jsonBytes, _ := json.Marshal(&transferJSON)
+	return jsonBytes
+}
+
+func generateGetBalanceJSON(targetAddress []byte) []byte {
+	// Encode the address as hex
+	hexTargetAddress := hex.EncodeToString(targetAddress)
+
+	getBalanceJSON := &gammacli.JSONTransaction{
+		ContractName: "BenchmarkToken",
+		MethodName:   "getBalance",
+		Arguments: []gammacli.JSONMethodArgument{
+			{Name: "targetAddress", Type: "bytes", Value: hexTargetAddress},
+		},
+	}
+
+	callJSONBytes, _ := json.Marshal(&getBalanceJSON)
+	return callJSONBytes
+}
+
+func generateGetCounterJSON() []byte {
+	getCounterJSON := &gammacli.JSONTransaction{
+		ContractName: "Counter",
+		MethodName:   "get",
+		Arguments:    []gammacli.JSONMethodArgument{},
+	}
+
+	callJSONBytes, _ := json.Marshal(&getCounterJSON)
+	return callJSONBytes
+}
+
+func generateAddCounterJSON(amount uint64) []byte {
+	arg := gammacli.JSONMethodArgument{Name: "amount", Type: "uint64", Value: amount}
+
+	addAmountToCounterJSON := &gammacli.JSONTransaction{
+		ContractName: "Counter",
+		MethodName:   "add",
+		Arguments:    []gammacli.JSONMethodArgument{arg},
+	}
+
+	addJSONBytes, _ := json.Marshal(&addAmountToCounterJSON)
+	return addJSONBytes
+}
+
+func TestGammaFlowWithActualJSONFilesUsingBenchmarkToken(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping e2e tests in short mode")
+	}
+
+	port := ":8080"
+	gamma := gammacli.StartGammaServer(port, false)
+	defer gamma.GracefulShutdown(1 * time.Second)
+
+	time.Sleep(100 * time.Millisecond) // wait for server to start
```

Comment:
this needs to go away as soon as we have a deterministic way to check if the server has started

Reply:
Agreed